arch/nios2/core/fatal.c

```diff
@@ -187,7 +187,7 @@ FUNC_NORETURN void _Fault(const NANO_ESF *esf)
 #if CONFIG_EXTRA_EXCEPTION_INFO
 	printk("reason: %s\n", cause_str(cause));
 #endif
-	if (BIT(cause) & NIOS2_BADADDR_CAUSE_MASK) {
+	if ((BIT(cause) & NIOS2_BADADDR_CAUSE_MASK) != 0) {
```

Comment:
It seems we could just use BIT(cause) == NIOS2_BADADDR_CAUSE_MASK here.

Reply:
no, NIOS2_BADADDR_CAUSE_MASK is more than one bit